Navigation Link, Navigation Submenu: Remove the title attribute controls

Infinite-Null opened the pull request:
Closes: #69663

## What?
Removes the Title attribute UI option from Navigation Submenu block and Custom Link button to align with accessibility best practices.

## Testing Instructions
1. Open a post or page. 
2. Insert an Image block.
3. Go to the Inspector control under the advanced section, and see if the title input is there.
4. Insert the Navigation block and add a submenu and a page link
5. For both blocks, check if the title input is there.

## Screenshots or screencast <!-- if applicable -->

<!-- If you would like to upload screenshots, feel free to use the table below when it is useful to show the difference between before and after the change. -->

|Before|After|
|-|-|
|<img width="280" alt="Screenshot 2025-03-27 at 11 41 08 AM" src="https://github.com/user-attachments/assets/8be556a4-3b3d-4e8d-98b7-3a3ac362d9d7" />|<img width="268" alt="Screenshot 2025-03-27 at 11 38 50 AM" src="https://github.com/user-attachments/assets/ec138e6b-d49a-4fd7-865c-55113d07258d" />|
|<img width="274" alt="Screenshot 2025-03-27 at 11 40 57 AM" src="https://github.com/user-attachments/assets/9cc41406-9fd8-446b-90d9-cdc5b6112011" />|<img width="268" alt="Screenshot 2025-03-27 at 11 39 07 AM" src="https://github.com/user-attachments/assets/fa601519-785a-4712-bfe8-4b661bf66901" />|

Infinite-Null committed:
Remove Title attribute controls

Infinite-Null committed:
Remove Title controls from Image component

Infinite-Null committed:
Remove 'title' attribute from navigation link and submenu blocks

Infinite-Null committed:
Remove title attribute checks from custom sources tests

Infinite-Null commented:
The failing tests don’t seem to be related to the changes in this PR.

Infinite-Null committed:
Merge branch 'trunk' into remove/title-attribute

t-hamano reviewed (commented):
Thanks for the PR! 

I think it would be good to focus only on two blocks, `core/navigation-link` and `core/navigation-submenu` in this PR.

If we plan to remove the `title` attribute from the Image block, we need to address the following tasks:

- If we update the save function, we will need to add [a block deprecation](https://github.com/WordPress/gutenberg/blob/trunk/docs/reference-guides/block-api/block-deprecation.md).
- The `title` attribute is allowed to be used as Block Binding API (See [this core code](https://github.com/t-hamano/wordpress-develop/blob/ed2f2ecd3fa043eca3a064a95fcb4830b5c0765c/src/wp-includes/class-wp-block.php#L288)). This means that we may need to prepare a patch that removes the `title` attribute from the supported attributes to the core along with this PR.

I think It's too big to tackle all the tasks in one PR.

Infinite-Null committed:
Revert image changes

Infinite-Null commented:
Hi @t-hamano,
I have updated the PR to remove `title` input from `core/navigation-submenu` and `core/navigation-link`. Could you please review this at your convenience.
Thank You!

t-hamano reviewed (commented):
I think we need to completely remove code related to the `title` attribute. For example:

https://github.com/WordPress/gutenberg/blob/d8545b29b5224207546d7b31dbd8502f804f439b/packages/block-library/src/navigation-link/index.php#L252-L254

https://github.com/WordPress/gutenberg/blob/d8545b29b5224207546d7b31dbd8502f804f439b/packages/block-library/src/navigation-link/transforms.js#L124

It's a good idea to go through the code in both blocks and look for anything that needs to be removed.

Infinite-Null committed:
Remove `title` attribute from navigation link and submenu blocks

Infinite-Null commented:
Hi @t-hamano,
I have updated the PR and removed the title attribute from other places as well. Could you please review it at your convenience and let me know if I’ve missed anything?

Thank You!

t-hamano reviewed (approved):
LGTM! I think all code related to the `title` attribute has been removed.

Since I think this change is very important for site builders, I'm adding a `Needs dev note` label to announce it in the next major release.

t-hamano commented:
@priethor As a follow-up to this PR, I recalled that the title attribute was restored internally in #70234. This means that only the title UI control was removed, so it doesn't affect backward compatibility. Therefore, I think a Dev Note is unnecessary.
